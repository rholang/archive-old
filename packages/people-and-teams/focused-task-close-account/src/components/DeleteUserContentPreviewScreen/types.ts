import { User } from '../../types';

export interface DeleteUserContentPreviewScreenProps {
  isCurrentUser: boolean;
  user: User;
  preferenceSelection: (username: string) => void;
}

export interface DeleteUserContentPreviewScreenState {
  currentActive: number;
}
