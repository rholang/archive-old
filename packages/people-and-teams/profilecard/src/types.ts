import * as React from 'react';
import { IntlShape } from 'react-intl';
export { WithOuterListenersProps } from './components/withOuterListeners';

export type Elevation = 'e100' | 'e200' | 'e300' | 'e400' | 'e500' | 'none';

export interface ApiClientResponse {
  User: {
    id: string;
    isBot: boolean;
    isCurrentUser: boolean;
    isNotMentionable: boolean;
    avatarUrl: string | null;
    email: string | null;
    fullName: string | null;
    location: string | null;
    meta: string | null;
    nickname: string | null;
    companyName: string | null;
    remoteTimeString: string | null;
    remoteWeekdayIndex: string | null;
    remoteWeekdayString: string | null;
    status: StatusType;
    statusModifiedDate: number | null;
  };
}

export interface ProfileCardClientData {
  isBot: boolean;
  isCurrentUser: boolean;
  isNotMentionable: boolean;
  avatarUrl?: string;
  email?: string;
  fullName?: string;
  location?: string;
  meta?: string;
  nickname?: string;
  companyName?: string;
  timestring?: string;
  status: StatusType;
  statusModifiedDate?: number | null;
}

export interface ProfileCardResourcedProps {
  userId: string;
  cloudId: string;
  resourceClient: ProfileClient;
  actions?: ProfileCardAction[];
  analytics?: any;
  position?: ProfilecardTriggerPosition;
  trigger?: TriggerType;
  children?: React.ReactNode;
  customElevation?: Elevation;
}

export interface ProfileCardResourcedState {
  visible?: boolean;
  isLoading?: boolean;
  hasError: boolean;
  error?: ProfileCardErrorType;
  data: ProfileCardClientData | null;
}

export interface ProfileCardTriggerProps {
  userId: string;
  cloudId: string;
  resourceClient: ProfileClient;
  actions?: ProfileCardAction[];
  analytics?: any;
  position?: ProfilecardTriggerPosition;
  trigger?: TriggerType;
  children?: React.ReactNode;
  customElevation?: Elevation;
}

export interface ProfileCardTriggerState {
  visible?: boolean;
  isLoading?: boolean;
  hasError: boolean;
  error?: ProfileCardErrorType;
  data: ProfileCardClientData | null;
}

export type StatusType = 'active' | 'inactive' | 'closed';

export type TriggerType = 'hover' | 'click';

export type StatusModifiedDateType =
  | 'noDate'
  | 'thisWeek'
  | 'thisMonth'
  | 'lastMonth'
  | 'aFewMonths'
  | 'severalMonths'
  | 'moreThanAYear';

export interface ProfileCardAction {
  callback?: (...args: any[]) => any;
  shouldRender?: (data: any) => boolean;
  id?: string;
  label: string;
}

export interface ProfilecardProps {
  isLoading?: boolean;
  hasError?: boolean;
  errorType?: ProfileCardErrorType;
  status?: StatusType;
  isBot?: boolean;
  isNotMentionable?: boolean;
  avatarUrl?: string;
  fullName?: string;
  meta?: string;
  // Nick name is also known as public name
  nickname?: string;
  email?: string;
  location?: string;
  companyName?: string;
  timestring?: string;
  actions?: ProfileCardAction[];
  clientFetchProfile?: any;
  analytics?: any;
  statusModifiedDate?: number | null;
  // allow to pass custom elevation, example value of this prop is: `e100`, `e200`, `e300`, `e400` and `e500`
  // Reference from `packages/core/theme/src/elevation.js` to see all valid values.
  customElevation?: Elevation;

  // Allow to pass custom message for disabled account which `status` prop is `inactive` or `closed`.
  // `disabledAccountMessage` should not contain react-intl components, ex: `FormattedMessage`,
  // because ProfileCard component is wrapped in its own `IntlProvider` and `FormattedMessage` will loads messages of `@atlaskit/profilecard`,
  // not from the consumer of `@atlaskit/profilecard`.
  disabledAccountMessage?: React.ReactNode;
  // Allow to show a status lozenge for disabled account which `status` prop is `inactive` or `closed`
  hasDisabledAccountLozenge?: boolean;
}

export interface MessageIntlProviderProps {
  children: React.ReactNode;
  intl: IntlShape;
}

export type RelativeDateKeyType =
  | 'ThisWeek'
  | 'ThisMonth'
  | 'LastMonth'
  | 'AFewMonths'
  | 'SeveralMonths'
  | 'MoreThanAYear'
  | null;

export interface ProfileClient {
  makeRequest: (
    cloudId: string,
    userId: string,
  ) => Promise<ProfileCardClientData>;
  setCachedProfile: (cloudId: string, userId: string, cacheItem: any) => void;
  getCachedProfile: (cloudId: string, userId: string) => ProfileCardClientData;
  flushCache: () => void;
  getProfile: (
    cloudId: string,
    userId: string,
  ) => Promise<ProfileCardClientData>;
}

export type ProfilecardTriggerPosition =
  | 'bottom-start'
  | 'bottom'
  | 'bottom-end'
  | 'left-start'
  | 'left'
  | 'left-end'
  | 'top-end'
  | 'top'
  | 'top-start'
  | 'right-end'
  | 'right'
  | 'right-start';

export type ProfileCardErrorType = {
  reason: 'default' | 'NotFound';
} | null;

export interface ProfileClientOptions {
  url: string;
  cacheSize?: number;
  cacheMaxAge?: number;
}

export interface ProfileClientConfig extends ProfileClientOptions {
  url: string;
  cacheSize: number;
  cacheMaxAge: number;
}
