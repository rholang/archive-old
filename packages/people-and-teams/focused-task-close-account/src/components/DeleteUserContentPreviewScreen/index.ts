export * from './DeleteUserContentPreviewScreen';
export { default } from './DeleteUserContentPreviewScreen';
