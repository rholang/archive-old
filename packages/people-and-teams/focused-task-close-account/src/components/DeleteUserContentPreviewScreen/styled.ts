import styled from 'styled-components';
import { borderRadius, colors, elevation, typography } from '@atlaskit/theme';
import gridSizeTimes from '../../util/gridSizeTimes';

interface SectionCardProps {
  isSelected: boolean;
}

const getSelectedCardColor = (props: SectionCardProps) => {
  return props.isSelected && `${colors.B50}`;
};

export const Screen = styled.div`
  width: 640px;
  margin-bottom: ${gridSizeTimes(4)}px;
`;

export const Title = styled.div`
  ${typography.h700};
  margin-bottom: ${gridSizeTimes(4)}px;
  margin-top: 0;
`;

export const SectionCard = styled.div`
  position: relative;
  display: flex;
  padding: ${gridSizeTimes(2.5)}px;
  width: 100%;
  background-color: ${(props: SectionCardProps) => getSelectedCardColor(props)};
  border-radius: ${borderRadius()}px;
  ${elevation.e200};
  margin-top: ${gridSizeTimes(2)}px;
`;

export const Avatar = styled.div`
  display: flex;
  flex-direction: column;
  margin-left: ${gridSizeTimes(2.5)}px;
  margin-right: ${gridSizeTimes(1)}px;
`;

export const UserDetails = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: ${gridSizeTimes(1.5)}px;
  font-weight: 600;
  color: ${colors.B400};
`;
