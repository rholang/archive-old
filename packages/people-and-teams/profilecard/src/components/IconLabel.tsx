import * as React from 'react';
import IconLocation from '@atlaskit/icon/glyph/location';
import IconRecent from '@atlaskit/icon/glyph/recent';
import IconEmail from '@atlaskit/icon/glyph/email';
import OfficeBuildingIcon from '@atlaskit/icon/glyph/office-building';

import {
  DetailsLabel,
  DetailsLabelIcon,
  DetailsLabelText,
} from '../styled/Card';

const icons = {
  location: IconLocation,
  time: IconRecent,
  email: IconEmail,
  companyName: OfficeBuildingIcon,
};

type Props = {
  icon: string;
  children?: React.ReactNode;
};

export default class IconLabel extends React.PureComponent<Props> {
  static defaultProps = {
    icon: '',
  };

  render() {
    if (!this.props.children) {
      return null;
    }

    // @ts-ignore
    const IconElement = this.props.icon && icons[this.props.icon];
    const displayIcon = IconElement ? (
      <IconElement label={`icon ${this.props.icon}`} size="small" />
    ) : null;

    return (
      <DetailsLabel>
        <DetailsLabelIcon>{displayIcon}</DetailsLabelIcon>
        <DetailsLabelText>{this.props.children}</DetailsLabelText>
      </DetailsLabel>
    );
  }
}
