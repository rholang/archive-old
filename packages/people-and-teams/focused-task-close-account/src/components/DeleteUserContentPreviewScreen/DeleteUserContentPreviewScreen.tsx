import * as React from 'react';
import { FormattedMessage, FormattedHTMLMessage } from 'react-intl';
import Avatar from '@atlaskit/avatar';

import { contentPreviewMessages } from '../../messages';
import * as Styled from './styled';
import MessagesIntlProvider from '../MessagesIntlProvider';
import {
  DeleteUserContentPreviewScreenProps,
  DeleteUserContentPreviewScreenState,
} from './types';

export class DeleteUserContentPreviewScreen extends React.Component<
  DeleteUserContentPreviewScreenProps,
  DeleteUserContentPreviewScreenState
> {
  state: DeleteUserContentPreviewScreenState = {
    currentActive: -1,
  };

  componentDidMount() {
    this.props.preferenceSelection('');
  }

  handleClickSection = (userName: string, position: number) => () => {
    this.props.preferenceSelection(userName);
    this.setState({ currentActive: position });
  };

  isCardSelected = (position: number): boolean => {
    return position === this.state.currentActive;
  };

  selectAdminOrSelfCopy = (
    adminCopy: FormattedMessage.MessageDescriptor,
    selfCopy: FormattedMessage.MessageDescriptor,
  ) => {
    return this.props.isCurrentUser ? selfCopy : adminCopy;
  };

  render() {
    const { user } = this.props;
    return (
      <MessagesIntlProvider>
        <Styled.Screen>
          <Styled.Title>
            <FormattedMessage
              {...this.selectAdminOrSelfCopy(
                contentPreviewMessages.headingAdmin,
                contentPreviewMessages.headingSelf,
              )}
            />
          </Styled.Title>
          <FormattedMessage
            {...this.selectAdminOrSelfCopy(
              contentPreviewMessages.paragraphSurveyAdmin,
              contentPreviewMessages.paragraphSurveySelf,
            )}
            tagName="p"
          />
          <FormattedMessage
            {...this.selectAdminOrSelfCopy(
              contentPreviewMessages.lineSurveyAdmin,
              contentPreviewMessages.lineSurveySelf,
            )}
            tagName="p"
          />
          <Styled.SectionCard
            className="nameSectionCard"
            onClick={this.handleClickSection('Name', 1)}
            isSelected={1 === this.state.currentActive}
          >
            <Styled.Avatar>
              <Avatar size="large" src="" />
            </Styled.Avatar>
            <Styled.UserDetails>{user.fullName}</Styled.UserDetails>
          </Styled.SectionCard>
          <Styled.SectionCard
            className="formerUserSectionCard"
            onClick={this.handleClickSection('Former User', 0)}
            isSelected={0 === this.state.currentActive}
          >
            <Styled.Avatar>
              <Avatar size="large" src="" />
            </Styled.Avatar>
            <Styled.UserDetails>
              <FormattedMessage {...contentPreviewMessages.formerUser} />
            </Styled.UserDetails>
          </Styled.SectionCard>
          <FormattedHTMLMessage
            {...this.selectAdminOrSelfCopy(
              contentPreviewMessages.footnoteAdmin,
              contentPreviewMessages.footnoteSelf,
            )}
            tagName="p"
          />
        </Styled.Screen>
      </MessagesIntlProvider>
    );
  }
}

export default DeleteUserContentPreviewScreen;
