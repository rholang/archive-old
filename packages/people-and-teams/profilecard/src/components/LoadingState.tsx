import * as React from 'react';
import AkSpinner from '@atlaskit/spinner';

import { SpinnerContainer } from '../styled/Card';

const LoadingState = () => (
  <SpinnerContainer>
    <AkSpinner />
  </SpinnerContainer>
);

export default LoadingState;
