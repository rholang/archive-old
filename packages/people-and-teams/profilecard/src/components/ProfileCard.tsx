import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import { colors } from '@atlaskit/theme';
import Avatar from '@atlaskit/avatar';
import Button from '@atlaskit/button';
import Lozenge from '@atlaskit/lozenge';
import Spinner from '@atlaskit/spinner';

import IconLabel from './IconLabel';
import ErrorMessage from './ErrorMessage';
import relativeDate from '../internal/relative-date';
import messages from '../messages';

import { ProfilecardProps } from '../types';

import {
  ActionButtonGroup,
  ActionsFlexSpacer,
  AppTitleLabel,
  CardContainer,
  CardContent,
  DisabledInfo,
  DetailsGroup,
  FullNameLabel,
  JobTitleLabel,
  ProfileImage,
  CardElevationWrapper,
  LozengeWrapper,
  SpinnerContainer,
} from '../styled/Card';
import { AnalyticsName } from '../internal/analytics';

export default class Profilecard extends React.PureComponent<ProfilecardProps> {
  static defaultProps: ProfilecardProps = {
    isLoading: false,
    hasError: false,
    errorType: null,
    status: 'active',
    isBot: false,
    isNotMentionable: false,
    actions: [],
    hasDisabledAccountLozenge: true,
    customElevation: 'e200',
    analytics: () => null,
    clientFetchProfile: () => null,
  };

  private timeOpen: number | null;
  clientFetchProfile: () => void;

  constructor(props: ProfilecardProps) {
    super(props);

    this.timeOpen = null;

    this.clientFetchProfile = (...args: any) => {
      this.callAnalytics(AnalyticsName.PROFILE_CARD_RELOAD, {});
      this.callClientFetchProfile(...args);
    };
  }

  private durationSince = (from: number | null) => {
    const fromParsed = from || 0;
    return fromParsed > 0 ? Date.now() - fromParsed : null;
  };

  callClientFetchProfile = (...args: any) => {
    if (this.props.clientFetchProfile) {
      this.props.clientFetchProfile(...args);
    }
  };

  callAnalytics = (id: string, options: any) => {
    if (this.props.analytics) {
      this.props.analytics(id, options);
    }
  };

  componentDidMount() {
    this.timeOpen = Date.now();
    this.callAnalytics(AnalyticsName.PROFILE_CARD_VIEW, {});
  }

  renderErrorMessage() {
    return (
      <ErrorMessage
        reload={this.props.clientFetchProfile && this.clientFetchProfile}
        errorType={this.props.errorType}
      />
    );
  }

  renderActionsButtons() {
    if (this.props.actions && this.props.actions.length === 0) {
      return null;
    }

    return (
      <ActionButtonGroup>
        {this.props.actions &&
          this.props.actions.map((action, idx) => (
            <Button
              appearance={idx === 0 ? 'default' : 'subtle'}
              key={action.label}
              onClick={(...args: any) => {
                this.callAnalytics(AnalyticsName.PROFILE_CARD_CLICK, {
                  id: action.id || null,
                  duration: this.durationSince(this.timeOpen),
                });
                if (action.callback) {
                  action.callback(...args);
                }
              }}
            >
              {action.label}
            </Button>
          ))}
      </ActionButtonGroup>
    );
  }

  renderCardDetailsDefault() {
    const { meta, location, email, timestring, companyName } = this.props;

    return (
      <DetailsGroup>
        {this.renderFullNameAndPublicName(meta)}
        {meta && <JobTitleLabel>{meta}</JobTitleLabel>}
        <IconLabel icon="email">{email}</IconLabel>
        <IconLabel icon="time">{timestring}</IconLabel>
        <IconLabel icon="companyName">{companyName}</IconLabel>
        <IconLabel icon="location">{location}</IconLabel>
      </DetailsGroup>
    );
  }

  renderCardDetailsForDisabledAccount() {
    const { status, companyName, hasDisabledAccountLozenge } = this.props;

    return (
      <DetailsGroup>
        <FullNameLabel noMeta isDisabledAccount>
          {this.getDisabledAccountName()}
        </FullNameLabel>

        {hasDisabledAccountLozenge && (
          <LozengeWrapper>
            <Lozenge appearance="default" isBold>
              {status === 'inactive' && (
                <FormattedMessage {...messages.inactiveAccountMsg} />
              )}
              {status === 'closed' && (
                <FormattedMessage {...messages.closedAccountMsg} />
              )}
            </Lozenge>
          </LozengeWrapper>
        )}

        <DisabledInfo>{this.getDisabledAccountDesc()}</DisabledInfo>

        {status === 'inactive' && (
          <IconLabel icon="companyName">{companyName}</IconLabel>
        )}
      </DetailsGroup>
    );
  }

  getDisabledAccountName() {
    const { nickname, fullName, status } = this.props;
    if (status === 'inactive') {
      return fullName || nickname;
    } else if (status === 'closed') {
      return (
        nickname || (
          <FormattedMessage {...messages.disabledAccountDefaultName} />
        )
      );
    }

    return null;
  }

  getDisabledAccountDesc() {
    const {
      status = 'closed',
      statusModifiedDate,
      disabledAccountMessage,
    } = this.props;
    const date = statusModifiedDate
      ? new Date(statusModifiedDate * 1000)
      : null;
    const relativeDateKey = relativeDate(date);

    // consumer does not want to use built-in message
    if (disabledAccountMessage) {
      return disabledAccountMessage;
    }

    let secondSentence: React.ReactNode = null;
    if (relativeDateKey) {
      secondSentence = (
        <FormattedMessage
          // @ts-ignore
          {...messages[`${status}AccountDescMsgHasDate${relativeDateKey}`]}
        />
      );
    } else {
      secondSentence = (
        // @ts-ignore
        <FormattedMessage {...messages[`${status}AccountDescMsgNoDate`]} />
      );
    }

    return (
      <p>
        <FormattedMessage {...messages.generalDescMsgForDisabledUser} />{' '}
        {secondSentence}
      </p>
    );
  }

  private renderFullNameAndPublicName(meta?: string) {
    const { nickname, fullName } = this.props;

    if (!fullName && !nickname) {
      return null;
    }

    const displayName =
      fullName === nickname
        ? fullName
        : `${fullName}${nickname ? ` (${nickname}) ` : ''}`;

    return <FullNameLabel noMeta={!meta}>{displayName}</FullNameLabel>;
  }

  renderCardDetailsApp() {
    return (
      <DetailsGroup>
        {this.renderFullNameAndPublicName()}
        <AppTitleLabel>App</AppTitleLabel>
      </DetailsGroup>
    );
  }

  renderCardDetails() {
    const { isBot, status } = this.props;

    if (isBot) {
      return this.renderCardDetailsApp();
    }

    if (status === 'inactive' || status === 'closed') {
      return this.renderCardDetailsForDisabledAccount();
    }

    return this.renderCardDetailsDefault();
  }

  render() {
    const { fullName, status, customElevation } = this.props;
    let cardContent: React.ReactNode = null;

    // @FIXME do closed users have empty fullName field?
    const canRender = fullName || status === 'closed';

    if (this.props.hasError) {
      this.callAnalytics(AnalyticsName.PROFILE_CARD_ERROR, {});

      cardContent = this.renderErrorMessage();
    } else if (this.props.isLoading) {
      cardContent = (
        <SpinnerContainer>
          <Spinner />
        </SpinnerContainer>
      );
    } else if (canRender) {
      const isDisabledUser = status === 'inactive' || status === 'closed';
      const actions = this.renderActionsButtons();

      this.callAnalytics(AnalyticsName.PROFILE_CARD_LOADED, {
        duration: this.durationSince(this.timeOpen),
      });

      cardContent = (
        <CardContainer isDisabledUser={isDisabledUser}>
          <ProfileImage>
            <Avatar
              size="xlarge"
              src={
                this.props.status !== 'closed'
                  ? this.props.avatarUrl
                  : undefined
              }
              borderColor={colors.N0}
            />
          </ProfileImage>
          <CardContent>
            {this.renderCardDetails()}
            {actions ? (
              <>
                <ActionsFlexSpacer />
                {actions}
              </>
            ) : null}
          </CardContent>
        </CardContainer>
      );
    }

    return (
      <CardElevationWrapper customElevation={customElevation}>
        {cardContent}
      </CardElevationWrapper>
    );
  }
}
