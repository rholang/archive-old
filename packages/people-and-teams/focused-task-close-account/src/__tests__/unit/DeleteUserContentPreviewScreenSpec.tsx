import * as React from 'react';
import { shallow } from 'enzyme';
import { DeleteUserContentPreviewScreen } from '../../components/DeleteUserContentPreviewScreen';
import { catherineHirons } from '../../mocks/users';
import { DeleteUserContentPreviewScreenProps } from '../../components/DeleteUserContentPreviewScreen/types';

const defaultProps: DeleteUserContentPreviewScreenProps = {
  isCurrentUser: false,
  user: catherineHirons,
  preferenceSelection: (name: string) => jest.fn(),
};

const render = (props = {}) =>
  shallow(<DeleteUserContentPreviewScreen {...defaultProps} {...props} />);

describe('DeleteUserContentPreviewScreen', () => {
  test('renders snapshot for personal accounts when isCurrentUser is true', () => {
    const wrapper = render({ isCurrentUser: true });
    expect(wrapper).toMatchSnapshot();
  });
  test('renders snapshot for unmanaged accounts when isCurrentUser is false', () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });
  test('componentDidMount calls the preferenceSelection prop', () => {
    const spyPreferenceSelection = jest.fn();
    render({ preferenceSelection: spyPreferenceSelection });
    expect(spyPreferenceSelection).toHaveBeenCalledTimes(1);
  });
});

describe('handleClickSelection', () => {
  test('changes the isSelected parameter of the element', () => {
    const wrapper = render();
    const divWrapper = wrapper.find('.nameSectionCard');
    divWrapper.simulate('click');
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
  test('calls "preferenceSelection" prop', () => {
    const preferenceSelection = jest.fn();
    const wrapper = render({ preferenceSelection });
    const divWrapper = wrapper.find('.nameSectionCard');
    divWrapper.simulate('click');
    wrapper.update();
    expect(preferenceSelection).toHaveBeenCalled();
  });
});
