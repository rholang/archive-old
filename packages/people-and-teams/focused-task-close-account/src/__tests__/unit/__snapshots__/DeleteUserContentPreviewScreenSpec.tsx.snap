// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteUserContentPreviewScreen renders snapshot for personal accounts when isCurrentUser is true 1`] = `
<InjectIntl(MessagesIntlProvider)>
  <styled.div>
    <styled.div>
      <FormattedMessage
        defaultMessage="How would you prefer your deleted account to appear?"
        description="Heading for the screen that explains how users would prefer to be remembered"
        id="focused-task-close-account.delete-account.content-preview.heading.self"
        values={Object {}}
      />
    </styled.div>
    <FormattedMessage
      defaultMessage="After your account is deleted, you’ll appear as “Former user” to other users. Please take a moment to answer our survey question."
      description="First paragraph for the survey for personal accounts"
      id="focused-task-close-account.delete-account.content-preview.paragraph.survey.self"
      tagName="p"
      values={Object {}}
    />
    <FormattedMessage
      defaultMessage="If we gave you a choice, how would you like to appear to other users?"
      description="Second paragraph for the survey for personal accounts"
      id="focused-task-close-account.delete-account.content-preview.line.survey.self"
      tagName="p"
      values={Object {}}
    />
    <styled.div
      className="nameSectionCard"
      isSelected={false}
      onClick={[Function]}
    >
      <styled.div>
        <WithAnalyticsContext(WithAnalyticsEvents(mapProps(withPseudoState(Avatar))))
          size="large"
          src=""
        />
      </styled.div>
      <styled.div>
        Catherine Hirons
      </styled.div>
    </styled.div>
    <styled.div
      className="formerUserSectionCard"
      isSelected={false}
      onClick={[Function]}
    >
      <styled.div>
        <WithAnalyticsContext(WithAnalyticsEvents(mapProps(withPseudoState(Avatar))))
          size="large"
          src=""
        />
      </styled.div>
      <styled.div>
        <FormattedMessage
          defaultMessage="Former user"
          description="The display name for a user when they do not select their name"
          id="focused-task-close-account.delete-account.content-preview.formerUser"
          values={Object {}}
        />
      </styled.div>
    </styled.div>
    <FormattedHTMLMessage
      defaultMessage="Note: Answering this survey will help us make a better experience for all users, including those deleting their accounts. You’ll still appear as “Former user” after you delete your account."
      description="Note to users that they would still be saved as Former User"
      id="focused-task-close-account.delete-account.content-preview.footnote.self"
      tagName="p"
      values={Object {}}
    />
  </styled.div>
</InjectIntl(MessagesIntlProvider)>
`;

exports[`DeleteUserContentPreviewScreen renders snapshot for unmanaged accounts when isCurrentUser is false 1`] = `
<InjectIntl(MessagesIntlProvider)>
  <styled.div>
    <styled.div>
      <FormattedMessage
        defaultMessage="How would you prefer the deleted user to appear?"
        description="Heading for the screen that explains admin would prefer user to be remembered"
        id="focused-task-close-account.delete-account.content-preview.heading.admin"
        values={Object {}}
      />
    </styled.div>
    <FormattedMessage
      defaultMessage="After the user’s account is deleted, they’ll appear as “Former user” to other users. Please take a moment to answer our survey question."
      description="First paragraph for the survey for admins"
      id="focused-task-close-account.delete-account.content-preview.paragraph.survey.admin"
      tagName="p"
      values={Object {}}
    />
    <FormattedMessage
      defaultMessage="If we gave you a choice, how would you like the deleted user to appear to other users?"
      description="Second paragraph for the survey for admins"
      id="focused-task-close-account.delete-account.content-preview.line.survey.admin"
      tagName="p"
      values={Object {}}
    />
    <styled.div
      className="nameSectionCard"
      isSelected={false}
      onClick={[Function]}
    >
      <styled.div>
        <WithAnalyticsContext(WithAnalyticsEvents(mapProps(withPseudoState(Avatar))))
          size="large"
          src=""
        />
      </styled.div>
      <styled.div>
        Catherine Hirons
      </styled.div>
    </styled.div>
    <styled.div
      className="formerUserSectionCard"
      isSelected={false}
      onClick={[Function]}
    >
      <styled.div>
        <WithAnalyticsContext(WithAnalyticsEvents(mapProps(withPseudoState(Avatar))))
          size="large"
          src=""
        />
      </styled.div>
      <styled.div>
        <FormattedMessage
          defaultMessage="Former user"
          description="The display name for a user when they do not select their name"
          id="focused-task-close-account.delete-account.content-preview.formerUser"
          values={Object {}}
        />
      </styled.div>
    </styled.div>
    <FormattedHTMLMessage
      defaultMessage="Note: Answering this survey will help us make a better experience for all users. The user will still appear as “Former user” after you delete their account."
      description="Note to admins that the users would still be saved as Former User"
      id="focused-task-close-account.delete-account.content-preview.footnote.admin"
      tagName="p"
      values={Object {}}
    />
  </styled.div>
</InjectIntl(MessagesIntlProvider)>
`;

exports[`handleClickSelection changes the isSelected parameter of the element 1`] = `
<InjectIntl(MessagesIntlProvider)>
  <styled.div>
    <styled.div>
      <FormattedMessage
        defaultMessage="How would you prefer the deleted user to appear?"
        description="Heading for the screen that explains admin would prefer user to be remembered"
        id="focused-task-close-account.delete-account.content-preview.heading.admin"
        values={Object {}}
      />
    </styled.div>
    <FormattedMessage
      defaultMessage="After the user’s account is deleted, they’ll appear as “Former user” to other users. Please take a moment to answer our survey question."
      description="First paragraph for the survey for admins"
      id="focused-task-close-account.delete-account.content-preview.paragraph.survey.admin"
      tagName="p"
      values={Object {}}
    />
    <FormattedMessage
      defaultMessage="If we gave you a choice, how would you like the deleted user to appear to other users?"
      description="Second paragraph for the survey for admins"
      id="focused-task-close-account.delete-account.content-preview.line.survey.admin"
      tagName="p"
      values={Object {}}
    />
    <styled.div
      className="nameSectionCard"
      isSelected={true}
      onClick={[Function]}
    >
      <styled.div>
        <WithAnalyticsContext(WithAnalyticsEvents(mapProps(withPseudoState(Avatar))))
          size="large"
          src=""
        />
      </styled.div>
      <styled.div>
        Catherine Hirons
      </styled.div>
    </styled.div>
    <styled.div
      className="formerUserSectionCard"
      isSelected={false}
      onClick={[Function]}
    >
      <styled.div>
        <WithAnalyticsContext(WithAnalyticsEvents(mapProps(withPseudoState(Avatar))))
          size="large"
          src=""
        />
      </styled.div>
      <styled.div>
        <FormattedMessage
          defaultMessage="Former user"
          description="The display name for a user when they do not select their name"
          id="focused-task-close-account.delete-account.content-preview.formerUser"
          values={Object {}}
        />
      </styled.div>
    </styled.div>
    <FormattedHTMLMessage
      defaultMessage="Note: Answering this survey will help us make a better experience for all users. The user will still appear as “Former user” after you delete their account."
      description="Note to admins that the users would still be saved as Former User"
      id="focused-task-close-account.delete-account.content-preview.footnote.admin"
      tagName="p"
      values={Object {}}
    />
  </styled.div>
</InjectIntl(MessagesIntlProvider)>
`;
